Build error on GCC 4.8.5

Compiler: gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44) 

```shell
/srt/haicrypt/cryspr.c: In function 'crysprFallback_MsEncrypt':
/srt/haicrypt/cryspr.c:473:6: error: 'for' loop initial declarations are only allowed in C99 mode
      for (size_t i = 0; i < sizeof(aad) / 4; ++i)
      ^
/srt/haicrypt/cryspr.c:473:6: note: use option -std=c99 or -std=gnu99 to compile your code
/srt/haicrypt/cryspr.c: In function 'crysprFallback_MsDecrypt':
/srt/haicrypt/cryspr.c:619:7: error: 'for' loop initial declarations are only allowed in C99 mode
       for (size_t i = 0; i < sizeof(aad) / 4; ++i)
```

Introduced in PR #2962.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1046,8 +1046,23 @@ macro(srt_set_stdcxx targetname spec)
 	endif()
 endmacro()
 
+macro(srt_set_stdc targetname spec)
+	set (stdcspec ${spec})
+	if (NOT "${stdcspec}" STREQUAL "")
+		if (CMAKE_VERSION VERSION_LESS "3.1")
+			target_compile_options(${targetname} PRIVATE -std=c${stdcspec})
+			message(STATUS "C STD: ${targetname}: forced C${stdcspec} standard - GNU option: -std=c${stdcspec}")
+		else()
+			set_target_properties(${targetname} PROPERTIES C_STANDARD ${stdcspec})
+			message(STATUS "C STD: ${targetname}: forced C${stdcspec} standard - portable way")
+		endif()
+	else()
+		message(STATUS "APP: ${targetname}: using default C standard")
+	endif()
+endmacro()
 
 srt_set_stdcxx(srt_virtual "${USE_CXX_STD_LIB}")
+srt_set_stdc(srt_virtual "99")
 
 set (VIRTUAL_srt $<TARGET_OBJECTS:srt_virtual>)
 

